### DISCUSSION TITLE
Does Rainbow key support if someone has keystore file?

### DISCUSSION THREAD

**Discussion by 0xbeny** [rainbow-me/rainbowkit#2240] [General]
pls let me know if you know any snippet that I can use. basically my users come from MEW browser wallet and they have keystore file in their hand.

**Comment by DanielSinclair** [comment]
No, RainbowKit is designed to connect to external wallets using standards like WalletConnect. Private keys should only ever be stored inside of a sandboxed wallet, and never utilized at the dApp level on the web.

**Reply by 0xbeny** [reply]
Thanks for your reply, I have already installed Rainbowkit, Is there any way to help the users who hold keystore files?
because if I want to try other ways, it will break my existing compatibility with Rainbowkit.

**Reply by DanielSinclair** [reply]
To access the MEW keystore file, you need to import it into the MEW Wallet. Then the user would connect to RainbowKit using the MEW button while the wallet is active. https://kb.myetherwallet.com/zh/security-and-privacy/what-is-a-keystore-file/
